src/server/editorServices.ts

```diff
@@ -2722,8 +2781,21 @@ namespace ts.server {
                         else {
                             // reload from the disk
                             this.reloadConfiguredProject(project, reason);
+                            // If this is solution, reload the project till the reloaded project contains the script info directly
+                            if (!project.containsScriptInfo(info) && project.isSolution()) {
+                                forEachResolvedProjectReferenceProject(
+                                    project,
+                                    child => {
+                                        if (!updatedProjects.has(child.canonicalConfigFilePath)) {
+                                            updatedProjects.set(child.canonicalConfigFilePath, true);
+                                            this.reloadConfiguredProject(child, reason);

```

Comment:
Should `reason` be updated to reflect that it's a child project of a solution?

Reply:
I think its fine.. Since its still loading project for that file and reason for now isnt specific to file or project